KishanJ-Patel/One-Stop-Shopper
# Add cart buttons and icons on product page

### Context

- This pull request addresses issue OSS-36 which describes a missing cart buttons bug on the product page.
- This bug makes it difficult for the user to know if the product is in the cart and control its quantity in the cart when visiting the product page.

### Changes

- Added an add-to-cart button under price of the product which is visible if the product is not in the cart.
- Added in-cart icon, quantity control button and total quantity number which are visible when product quantity in the cart is at least 1.

### Testing

- To test this bugfix:
  1. Open the home page of the web app.
  2. Click on a product card to open its product page.
  3. Observe the section under the price.
  4. Click on the add-to-cart button to add the product to the cart.
  5. Click the quantity control buttons, plus and minus, to control the quantity of the product in the cart.

### Related Issues

- Closes `OSS-36`

### Reviewer Requests

- @KishanJ-Patel : For frontend review
- @KishanJ-Patel : For backend review
- @KishanJ-Patel : For testing and QA
